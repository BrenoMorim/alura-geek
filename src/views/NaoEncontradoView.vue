<template>
  <section class="nao-encontrado">
    <h2 class="nao-encontrado__titulo">Nada foi encontrado =(</h2>
    <h3 class="nao-encontrado__subtitulo">Erro 404</h3>
    <a class="botao-voltar" @click="this.$router.back()">Voltar</a>
  </section>
</template>

<script>
import { defineComponent } from "@vue/runtime-core";

export default defineComponent({
    name: 'NaoEncontradoView'
})
</script>

<style scoped>
.nao-encontrado {
    padding: 2rem 8rem;
    color: var(--cinza-escuro);
    background-color: var(--cinza-claro);
    text-align: center;
}
.nao-encontrado__titulo {
    font-size: 2.25rem;
    font-weight: 700;
    margin-bottom: 2rem;
}
.nao-encontrado__subtitulo {
    font-size: 2rem;
    color: var(--cinza);
    font-weight: 400;
    margin-bottom: 2rem;
}
.botao-voltar {
    background-color: var(--azul);
    color: var(--branco);
    padding: .5rem 3rem;
    font-size: 1.5rem;
    font-weight: 300;
    border-radius: 1rem;
    font-family: var(--fonte-principal);
    text-decoration: none;
    cursor: pointer;
}
</style>