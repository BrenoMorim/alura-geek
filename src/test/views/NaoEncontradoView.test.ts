import { expect, describe, it } from "vitest";
import { mount } from '@vue/test-utils';
import NaoEncontradoView from '../../views/NaoEncontradoView.vue';

describe('Na página de não encontrado', () => {

    const wrapper = mount(NaoEncontradoView); 

    it('Deve ser renderizada', () => {
        expect(wrapper.isVisible()).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });

    it('Deve conter mensagem e botão para voltar', () => {
        expect(wrapper.get('.nao-encontrado__titulo').text()).toBe('Nada foi encontrado =(');
        expect(wrapper.get('.nao-encontrado__subtitulo').text()).toBe('Erro 404');
        
        const botaoVoltar = wrapper.get('.botao-voltar');
        expect(botaoVoltar.text()).toBe('Voltar');
        expect(botaoVoltar.attributes('disabled')).toBeFalsy();
    });

});
