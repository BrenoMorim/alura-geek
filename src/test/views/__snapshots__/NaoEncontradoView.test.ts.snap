// Vitest Snapshot v1

exports[`Na página de não encontrado > Deve ser renderizada 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <section
        class="nao-encontrado"
        data-v-4fdff3bb=""
      >
        <h2
          class="nao-encontrado__titulo"
          data-v-4fdff3bb=""
        >
          Nada foi encontrado =(
        </h2>
        <h3
          class="nao-encontrado__subtitulo"
          data-v-4fdff3bb=""
        >
          Erro 404
        </h3>
        <a
          class="botao-voltar"
          data-v-4fdff3bb=""
        >
          Voltar
        </a>
      </section>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "hasOwnProperty": [Function],
          },
          "da": null,
          "data": {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {
            "hasOwnProperty": [Function],
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Nada foi encontrado =(",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h2
                  class="nao-encontrado__titulo"
                  data-v-4fdff3bb=""
                >
                  Nada foi encontrado =(
                </h2>,
                "key": null,
                "patchFlag": -1,
                "props": {
                  "class": "nao-encontrado__titulo",
                },
                "ref": null,
                "scopeId": "data-v-4fdff3bb",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h2",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Erro 404",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h3
                  class="nao-encontrado__subtitulo"
                  data-v-4fdff3bb=""
                >
                  Erro 404
                </h3>,
                "key": null,
                "patchFlag": -1,
                "props": {
                  "class": "nao-encontrado__subtitulo",
                },
                "ref": null,
                "scopeId": "data-v-4fdff3bb",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h3",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Voltar",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <a
                  class="botao-voltar"
                  data-v-4fdff3bb=""
                >
                  Voltar
                </a>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "botao-voltar",
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": "data-v-4fdff3bb",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "a",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [],
            "dynamicProps": null,
            "el": <section
              class="nao-encontrado"
              data-v-4fdff3bb=""
            >
              <h2
                class="nao-encontrado__titulo"
                data-v-4fdff3bb=""
              >
                Nada foi encontrado =(
              </h2>
              <h3
                class="nao-encontrado__subtitulo"
                data-v-4fdff3bb=""
              >
                Erro 404
              </h3>
              <a
                class="botao-voltar"
                data-v-4fdff3bb=""
              >
                Voltar
              </a>
            </section>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "nao-encontrado",
            },
            "ref": null,
            "scopeId": "data-v-4fdff3bb",
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "section",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/home/breno/Studies/front-end/alura-geek/src/views/NaoEncontradoView.vue",
            "__scopeId": "data-v-4fdff3bb",
            "components": {},
            "name": "NaoEncontradoView",
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <section
          class="nao-encontrado"
          data-v-4fdff3bb=""
        >
          <h2
            class="nao-encontrado__titulo"
            data-v-4fdff3bb=""
          >
            Nada foi encontrado =(
          </h2>
          <h3
            class="nao-encontrado__subtitulo"
            data-v-4fdff3bb=""
          >
            Erro 404
          </h3>
          <a
            class="botao-voltar"
            data-v-4fdff3bb=""
          >
            Voltar
          </a>
        </section>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/home/breno/Studies/front-end/alura-geek/src/views/NaoEncontradoView.vue",
          "__scopeId": "data-v-4fdff3bb",
          "components": {},
          "name": "NaoEncontradoView",
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <section
          class="nao-encontrado"
          data-v-4fdff3bb=""
        >
          <h2
            class="nao-encontrado__titulo"
            data-v-4fdff3bb=""
          >
            Nada foi encontrado =(
          </h2>
          <h3
            class="nao-encontrado__subtitulo"
            data-v-4fdff3bb=""
          >
            Erro 404
          </h3>
          <a
            class="botao-voltar"
            data-v-4fdff3bb=""
          >
            Voltar
          </a>
        </section>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {
    "hasOwnProperty": [Function],
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <section
    class="nao-encontrado"
    data-v-4fdff3bb=""
  >
    <h2
      class="nao-encontrado__titulo"
      data-v-4fdff3bb=""
    >
      Nada foi encontrado =(
    </h2>
    <h3
      class="nao-encontrado__subtitulo"
      data-v-4fdff3bb=""
    >
      Erro 404
    </h3>
    <a
      class="botao-voltar"
      data-v-4fdff3bb=""
    >
      Voltar
    </a>
  </section>,
}
`;
